app/src/main/java/net/gsantner/markor/frontend/textview/HighlightingEditor.java

```diff
@@ -139,33 +140,42 @@ private void updateHighlighting(final boolean recompute) {
             // Don't highlight unless shifted sufficiently or a recompute is required
             if (recompute || (visible && _hl.hasSpans() && isScrollSignificant())) {
 
+                final boolean heightSame = _hlRect.height() == _oldHlRect.height();
+
                 // Addition of spans which require reflow can shift text on re-application of spans
                 // we compute the resulting shift and scroll the view to compensate in order to make
-                // the experience smooth for the user
-                final int shiftTestLine = layout.getLineForVertical(_hlRect.centerY());
-                final int oldOffset = layout.getLineBaseline(shiftTestLine);
+                // the experience smooth for the user.
+                int shiftTestLine = -1, oldOffset = -1;
+                if (_scrollView != null && heightSame) {
+                    shiftTestLine = layout.getLineForVertical(_hlRect.centerY());
+                    oldOffset = layout.getLineBaseline(shiftTestLine);
+                }
+
+                if (heightSame) {
+                    // Hack to block bring point into view
+                    // We don't call this when height is changing
+                    blockBringPointIntoView();
```

Comment:
Refined how the blocking is done. Doing this too much is what lead to the original bug where the cursor would scroll away.